storage/src/macros.rs

```diff
@@ -0,0 +1,29 @@
+
+/// Macro to register and use a metric with description and labels.
+/// This macro is a wrapper around the `metrics` crate's `counter!` and `describe_counter!`
+/// macros. It ensures that the description is registered only once.
+///
+/// Usage:
+///   firewood_metric!("metric_name", "description", value_to_increment_by)
+///   firewood_metric!("metric_name", "description", value_to_increment_by, "label" => "value")
+#[macro_export]
+macro_rules! firewood_metric {
+    // With labels
+    ($name:expr, $desc:expr, $inc:expr, $($labels:tt)+) => {
+        {
+            static METRIC: std::sync::LazyLock<()> = std::sync::LazyLock::new(|| {
+                metrics::describe_counter!($name, $desc);
+            });
+            metrics::counter!($name, $($labels)+).increment($inc);
```

Comment:
Let's leave off the call to `increment($inc)` and have this just be a wrapper around `metrics::describe_counter` and `metrics::counter` and leave calling `.increment` to the caller, like the metrics macros do. 

That way we don't force everyone to use increment when it's not appropriate, such as timestamps/timedeltas where `.absolute` is more appropriate.